roofit/histfactory/src/ParamHistFunc.cxx

```diff
@@ -166,11 +166,10 @@ Int_t ParamHistFunc::GetNumBins( const RooArgSet& vars ) {
   RooAbsArg* comp ;
   while((comp = (RooAbsArg*) varIter.next())) {
     if (!dynamic_cast<RooRealVar*>(comp)) {
-      std::cout << "ParamHistFunc::GetNumBins" << vars.GetName() << ") ERROR: component " 
-	   << comp->GetName() 
-	   << " in vars list is not of type RooRealVar" << std::endl ;
-      RooErrorHandler::softAbort() ;
-      return -1;
+      auto errorMsg = std::string("ParamHistFunc::GetNumBins") + vars.GetName() + ") ERROR: component "
+                      + comp->GetName() + " in vars list is not of type RooRealVar";
+      std::cout << errorMsg << std::endl;
```

Comment:
`coutE` as well?

Reply:
It unfortunately doesn't work in a static function. Do you have an alternative suggestion? There is `oocoutE`, but I'm not sure if it makes sense because you have to give it a caller.

Reply:
A `static_cast<TObject*>(nullptr)` works fine. The caller is only necessary to print its name. If it's `nullptr`, that name printing is skipped.